visualize-admin/visualization-tool
# feat: Improve sorting for mix of hierarchical and regular values

Fixes #1015 (sorting issue).

## Goals / Scope

The main goal of this PR is to improve sorting of hierarchical dimensions (X field) in column charts.

## Description

As mentioned in #1015, we should first sort by hierarchical values and then by non-hierarchical values (by depth). We also fall back to sorting by label in `byAuto` mode, as it seems that sometimes `identifier` and `position` are undefined.

This required us to fetch components with hierarchies, but this change is only scoped to column charts, for other charts we do not need to fetch hierarchies for this reason.

Additionally, I refactored GQL queries, so we do not fetch dimensions & measures in `DataCubeObservations` query, but we need to explicitly fetch them with `Components` query. I believe it makes it more clear (and potentially efficient) to separate this logic.

## How to test

1. Go to [/en/browse?dataset=https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fnfi%2FC-616%2Fcube%2F1&dataSource=Int](https://visualization-tool-git-feat-improve-sorting-ixt1.vercel.app/en/browse?previous=%7B%22includeDrafts%22%3Atrue%2C%22order%22%3A%22SCORE%22%2C%22search%22%3A%22target%22%7D&dataset=https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fnfi%2FC-616%2Fcube%2F1&dataSource=Int).
2. Change vertical axis to `Volume (relativ)`.
3. Change horizontal axis to `Region`.
4. See that the bars are sorted in "logical" order (by hierarchy depth) – country is first (Switzerland), followed by smaller units (production & economics regions). Cantons are occupying the last places, as they were not part of the hierarchy.
